tests/access/test_AccessControl.py

```diff
@@ -0,0 +1,178 @@
+import pytest
+from pathlib import Path
+from starkware.starknet.testing.starknet import Starknet
+from utils import (
+    TestSigner, TRUE, FALSE,
+    assert_event_emitted, assert_revert,
+    get_contract_def, cached_contract
+)
+
+DEFAULT_ADMIN_ROLE = 0
+SOME_OTHER_ROLE = 42
+
+signer = TestSigner(123456789987654321)
+
+
+@pytest.fixture(scope='module')
+def contract_defs():
+    return {
+        Path(key).stem: get_contract_def(key)
+        for key in [
+            'openzeppelin/account/Account.cairo',
+            'tests/mocks/AccessControl.cairo',
+        ]
+    }
+
+
+@pytest.fixture(scope='module')
+async def accesscontrol_init(contract_defs):
+    starknet = await Starknet.empty()
+    account1 = await starknet.deploy(
+        contract_def=contract_defs['Account'],
+        constructor_calldata=[signer.public_key]
+    )
+    account2 = await starknet.deploy(
+        contract_def=contract_defs['Account'],
+        constructor_calldata=[signer.public_key]
+    )
+    accesscontrol = await starknet.deploy(
+        contract_def=contract_defs['AccessControl'],
+        constructor_calldata=[account1.contract_address]
+    )
+    return starknet.state, accesscontrol, account1, account2
+
+
+@pytest.fixture
+def accesscontrol_factory(contract_defs, accesscontrol_init):
+    state, accesscontrol, account1, account2 = accesscontrol_init
+    _state        = state.copy()
+    accesscontrol = cached_contract(_state, contract_defs['AccessControl'], accesscontrol)
+    account1      = cached_contract(_state, contract_defs['Account'], account1)
+    account2      = cached_contract(_state, contract_defs['Account'], account2)
+    return accesscontrol, account1, account2
+
+
+@pytest.mark.asyncio
+async def test_constructor(accesscontrol_factory):
+    accesscontrol, account1, account2 = accesscontrol_factory
+
+    expected1 = await accesscontrol.hasRole(DEFAULT_ADMIN_ROLE, account1.contract_address).call()
+    assert expected1.result.hasRole == TRUE
+
+    expected2 = await accesscontrol.hasRole(DEFAULT_ADMIN_ROLE, account2.contract_address).call()
+    assert expected2.result.hasRole == FALSE
+
+    expected3 = await accesscontrol.getRoleAdmin(DEFAULT_ADMIN_ROLE).call()
+    assert expected3.result.admin == DEFAULT_ADMIN_ROLE
+
+
+@pytest.mark.asyncio
+async def test_grantRole(accesscontrol_factory):
+    accesscontrol, account1, account2 = accesscontrol_factory
+
+    tx_exec_info = await signer.send_transaction(account1, accesscontrol.contract_address, 'grantRole', [0, account2.contract_address])
+    assert_event_emitted(
+        tx_exec_info,
+        from_address=accesscontrol.contract_address,
+        name='RoleGranted',
+        data=[
+            DEFAULT_ADMIN_ROLE,        # role
+            account2.contract_address, # account
+            account1.contract_address  # sender
+        ]
+    )
+    expected = await accesscontrol.hasRole(DEFAULT_ADMIN_ROLE, account2.contract_address).call()
```

Comment:
```suggestion
    expected = await accesscontrol.hasRole(DEFAULT_ADMIN_ROLE, account2.contract_address).invoke()
```
@andrew-fleming can correct me but invoking for some reason is faster than calling while testing?